lib/common/strings.c

```diff
@@ -148,6 +150,129 @@ crm_parse_int(const char *text, const char *default_text)
     return (int) result;
 }
 
+/*!
+ * \internal
+ * \brief Scan a double-precision floating-point value from a string
+ *
+ * \param[in]  text      String to scan
+ * \param[out] result    If not NULL, where to store scanned value
+ * \param[out] end_text  If not NULL, where to store pointer to just
+ *                       after value
+ *
+ * \return Standard Pacemaker return code (also set errno on error)
+ */
+static int
+pcmk__scan_double(const char *text, double *result, char **end_text)
```

Comment:
Should static functions use the "pcmk__" prefix or not (since there's no possibility of name conflicts outside the file)? I noticed that the scan_ll() name was set only 6 months ago as a best-practices commit: fc94233e.

Reply:
It's not terribly important either way, but I prefer not. It's not so obvious at the moment since we have so many legacy names, but eventually it would be: pcmk_ = public, pcmk__ = private, none = static, which would be a quick visual clue when you're browsing code. Also, as you mentioned, the prefixes are needed primarily to avoid name collisions, which is not an issue with static functions and just makes their names unnecessarily longer.

Reply:
Ack, next push will use scan_double().

Reply:
Next push will also update the macros in the pcmk_parse_double() unit test to remove "PCMK__", since they're local and not in a header file.